eras/conway/impl/src/Cardano/Ledger/Conway/TxInfo.hs

```diff
@@ -46,19 +56,44 @@ import Data.Foldable (Foldable (..))
 import qualified Data.Map.Strict as Map
 import qualified Data.Set as Set
 import Data.Text (Text)
+import GHC.Generics (Generic)
 import Lens.Micro
 import qualified PlutusLedgerApi.V3 as PV3
 
--- TODO implement this once Plutus releases an API with new certs
+data ConwayScriptPurpose era
+  = Minting !(PolicyID (EraCrypto era))
+  | Spending !(TxIn (EraCrypto era))
+  | Rewarding !(RewardAcnt (EraCrypto era))
+  | Certifying !(TxCert era)
+  | Voting !(VotingProcedure era)
```

Comment:
In order to avoid name collisions.
```suggestion
  = ConwayScriptMinting !(PolicyID (EraCrypto era))
  | ConwayScriptSpending !(TxIn (EraCrypto era))
  | ConwayScriptRewarding !(RewardAcnt (EraCrypto era))
  | ConwayScriptCertifying !(TxCert era)
  | ConwayScriptVoting !(VotingProcedure era)
```

If you feel like renaming ones from Alonzo too with deprecation pattern synonyms, that would be great, but we can do it at a later point.